src/main/scala/sigmastate/interpreter/Interpreter.scala

```diff
@@ -93,6 +94,17 @@ trait Interpreter extends ScorexLogging {
       else
         null
 
+    case t: GetVar[_] =>
+      if (context.extension.values.contains(t.varId)) {
+        val v = context.extension.values(t.varId)
+        if (v.tpe != t.tpe.elemType)
+          NoneValue(t.tpe.elemType)
```

Comment:
@kushti Should we throw an exception if context variable is defined but has wrong type? /cc @greenhat

Reply:
@ergomorphic @kushti Let me fix it, then. To be sure I get it, we want `getVar[Int](1).isDefined` to throw an exception if the defined variable is not Int, right?

Reply:
@ergomorphic @kushti Implemented throwing on invalid type in both context variable and register extraction.